### ISSUE TITLE
Failing to generate import in certain situations

### ISSUE THREAD

**Issue by datagutt** [chrishoermann/zod-prisma-types#180]
## Description of issue
I have some issues with certain files not importing schemas before using them.

## Dependencies
- prisma 5.1.0
- zod-prisma-types 2.7.9
## Code
I have the following schema:
```prisma
generator zod {
  provider                  = "zod-prisma-types"
  output                    = "./generated/schemas" // default is ./generated/zod
  useMultipleFiles          = true // default is false
  createModelTypes          = false // default is true
  createRelationValuesTypes = true // default is false
}

/// @@Gen.model(hide: true)
model Role {
  id        String    @id @default(uuid())
  createdAt DateTime  @default(now())
  updatedAt DateTime? @updatedAt
  deletedAt DateTime?

  PermissionForRole PermissionForRole[]
}

/// @@Gen.model(hide: true)
model PermissionForRole {
  id        String    @id @default(uuid())
  createdAt DateTime  @default(now())
  updatedAt DateTime  @updatedAt
  deletedAt DateTime?

  roleId String
  role   Role   @relation(fields: [roleId], references: [id])

  @@unique([roleId])
}
```

This full reproduction repo can be found at https://github.com/datagutt/prisma-zod-bug.
## Current behaviour
In generated/schemas/outputTypeSchemas/RoleFindManyArgsSchema.ts, this line fails:
```typescript

  PermissionForRole: z.union([z.boolean(),z.lazy(() => PermissionForRoleFindManyArgsSchema)]).optional(),
```
This is due to ``PermissionForRoleFindManyArgsSchema`` never being imported. This seems to only be happening in the ``RoleFindManyArgsSchema.ts``-file, not other files.

## Expected behaviour
``PermissionForRoleFindManyArgsSchema`` should be imported in the files that use it.

## Additional information
It seems that this bug is not triggered if the two models are named something completely different. I think the bug is related to both models starting or ending with "Role". Renaming ``PermissionForRole``to something different causes the file to be correctly imported.

**Comment by datagutt** [comment]
Seems that [this workaround](https://github.com/chrishoermann/zod-prisma-types/blob/96c7dcb5c217baec88204922207f4e8899892f79/packages/generator/src/functions/contentWriters/writeOutputObjectType.ts#L31C1-L36) is what causes the import to disappear.
If i remove the .filter(), it starts working again. Hopefully this helps you debug the issue.

**Comment by datagutt** [comment]
Oh, i see. The .includes() part is to avoid self-reference, but it doesn't actually check if the import starts with ``/``, causing ``PermissionForRoleFindManyArgsSchema`` to match, since it contains ``RoleFindManyArgsSchema``.
I have a fix for this in #181.

Have a nice day!
